app/controllers/users_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UsersController < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/users/new' do
+    erb(:"users/new")
+  end
+
+  post '/users' do
+    user = User.create(email: params[:email], password: params[:password], username: params[:username], name: params[:name])
+    if user == 1
+      flash[:notice] = 'Email already exists.'
```

Comment:
I would maybe pull the responsibility of checking if the email exists etc. into the user model, returning the appropriate string to flash[:notice], and then only having the if/else statement worry about the redirects - we try to push as much of the business logic into the model as possible